R/integrate_sces.R

```diff
@@ -172,17 +187,28 @@ integrate_harmony <- function(merged_sce,
     stop("The provided covariate columns are not all present in the `merged_sce` colData.")
   }
 
+  # check that the lambda values match up
+  if(length(covariate_cols) != length(covariate_lambda)){
+    stop("The number of covariate columns must be equal to the number of covariate lambda values.")
+  }
```

Comment:
I like having the option here, but maybe we can set it automatically to 1 if not set at all? Which is to say if `length(covariate_lambda) == 0` we could set `covariate_lambda = rep(1, length(covariate_cols))`? Maybe with a warning? I also don't know how sensitive things are to this parameter. Maybe there is another default that would be better?

If all 1s is acceptable, I would  actually characterize this as their bug, but that is neither here nor there.

Reply:
Given we haven't done much testing of this parameter at all, I would favor using the recommended default of 1, which is what we were previously doing by not setting it. I do like the idea of setting it, if it isn't provided though, so I'll add that check in.